src/nle.c

```diff
@@ -26,8 +28,107 @@
 
 extern int unixmain(int, char **);
 
+signed char
+vt_char_color_extract(TMTCHAR *c)
+{
+    /* We pick out the colors in the enum tmt_color_t. These match the order
+     * found standard in IBM color graphics, and are the same order as those
+     * found in src/color.h. We take the values from color.h, and choose
+     * default to be bright black (NO_COLOR) as nethack does.
+     *
+     * Finally we indicate whether the color is reverse, by indicating the
+     * sign
+     * of the final integer.
+     */
+    signed char color = 0;
+    switch (c->a.fg) {
```

Comment:
There's no need to do it this way.

You can just do `(c->a.bold) ? BRIGHT | c->a.gf : c->a.gf` in all cases, where `BRIGHT` should be defined in NetHack and otherwise `#define BRIGHT 8`.

Reply:
TMT_COLOR_X does not line up with nethacks CLR_X - they are off by one.  As long as it is, this way is more explicit and probably more 'done properly'.